x-pack/plugins/fleet/server/types/models/agent_policy.ts

```diff
@@ -84,6 +84,53 @@ export const AgentPolicyBaseSchema = {
   ),
   ...getSettingsAPISchema('AGENT_POLICY_ADVANCED_SETTINGS'),
   supports_agentless: schema.maybe(schema.boolean({ defaultValue: false })),
+  global_data_tags: schema.maybe(
+    schema.arrayOf(
+      schema.object({
+        name: schema.string(),
+        value: schema.oneOf([schema.string(), schema.number()]),
+      }),
+      {
+        validate: (tags) => {
```

Comment:
could we add unit tests on the validate function?

Reply:
++ on what Julia requested. As the validation logic is not straightforward, I would extract it on its own function and add unit tests.